vertigo-bot-designer/src/main/java/io/vertigo/chatbot/designer/builder/services/WelcomeTourStepServices.java

```diff
@@ -0,0 +1,122 @@
+import io.vertigo.datamodel.criteria.Criterions;
+import io.vertigo.datamodel.structure.model.DtList;
+import io.vertigo.datamodel.structure.model.DtListState;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static io.vertigo.chatbot.designer.utils.ListUtils.MAX_ELEMENTS_PLUS_ONE;
+
+@Transactional
+public class WelcomeTourStepServices implements Component {
+
+	@Inject
+	private WelcomeTourStepDAO welcomeTourStepDAO;
+
+	/**
+	 * Regex to extract params from welcome tour step config file
+	 * Ex : #[start.title Bienvenue]#", "#[start.text Bienvenue sur notre site de démo. Nous allons vous guider !]#
+	 * Group 1 is the step key, group 2 the param and group 3 the default value
+	 */
+	private static final Pattern extractPattern = Pattern.compile("#\\[([A-Za-z1-9\\-]+)\\.([A-Za-z1-9\\-]+)\\s+(.+)\\]#");
+
+	public WelcomeTourStep findById(final long tourStepId) {
+		return welcomeTourStepDAO.get(tourStepId);
+	}
+
+	public void delete(final long tourStepId) {
+		welcomeTourStepDAO.delete(tourStepId);
+	}
+
+	public void deleteAllByTourId(final long tourId) {
+		findAllStepsByTourId(tourId).forEach(welcomeTourStep -> delete(welcomeTourStep.getWelStepId()));
+	}
+
+	public WelcomeTourStep save (final WelcomeTourStep welcomeTourStep) {
+		return welcomeTourStepDAO.save(welcomeTourStep);
+	}
+
+	public DtList<WelcomeTourStep> findAllStepsByTourId(final long tourId) {
+		return welcomeTourStepDAO.findAll(Criterions.isEqualTo(DtDefinitions.WelcomeTourStepFields.tourId, tourId), DtListState.of(MAX_ELEMENTS_PLUS_ONE));
+	}
+
+	public Optional<WelcomeTourStep> findStepByTourIdAndStepId(final long tourId, final String stepId) {
+		return welcomeTourStepDAO.findOptional(Criterions.isEqualTo(DtDefinitions.WelcomeTourStepFields.tourId, tourId)
+				.and(Criterions.isEqualTo(DtDefinitions.WelcomeTourStepFields.internalStepId, stepId)));
+	}
+
+	public List<WelcomeTourStep> readStepsFromConfigString(final String config) {
+		final List<WelcomeTourStep> steps = new ArrayList<>();
+		final Map<String, Map<String, String>> parsed = parseFile(config);
+		Long sequence = 1L;
+		for (final Map.Entry<String, Map<String, String>> entry : parsed.entrySet()) {
+			final WelcomeTourStep welcomeTourStep = new WelcomeTourStep();
+			final Map<String, String> attributes = entry.getValue();
+			welcomeTourStep.setInternalStepId(entry.getKey());
+			if (attributes.get("title") != null) {
+				welcomeTourStep.setTitle(attributes.get("title"));
+			} else {
+				throw new VSystemException("Missing 'title' attribute on step '" + entry.getKey() + "'");
+			}
+			if (attributes.get("text") != null) {
+				welcomeTourStep.setText(attributes.get("text"));
+			} else {
+				throw new VSystemException("Missing 'text' attribute on step '" + entry.getKey() + "'");
+			}
```

Comment:
moins d'imbrications et l'erreur est utilisateur et non système
```suggestion
			if (!attributes.containsKey("title")) {
				throw new VUserException("Missing 'title' attribute on step '{0}'", entry.getKey());
			}
			welcomeTourStep.setTitle(attributes.get("title"));
			if (!attributes.containsKey("text")) {
				throw new VUserException("Missing 'text' attribute on step '{0}'", entry.getKey());
			}
			welcomeTourStep.setText(attributes.get("text"));
```